### PR TITLE
Fix CA1805 missing trivia

### PR DESCRIPTION
**Pull request by wzchua**
Fixes https://github.com/dotnet/roslyn-analyzers/issues/5412

### CODE REVIEW

File: src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarilyTests.cs [hunk 1]
```diff
@@ -255,5 +257,82 @@ public async Task Diagnostics_VB()
...
+                    .WithLocation(1),
+                VerifyCS.Diagnostic(DoNotInitializeUnnecessarilyAnalyzer.DefaultRule)
+                    .WithArguments("SomeIntProp2")
+                    .WithLocation(2),
+                VerifyCS.Diagnostic(DoNotInitializeUnnecessarilyAnalyzer.DefaultRule)
+                    .WithArguments("SomeIntProp3")
+                    .WithLocation(3));
+        }
+
+        private static VerifyCS.Test GetTestCS(string source, string corrected)
```
Comment (NewellClark) [inline src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarilyTests.cs:315]:
```suggestion
        }
        return;
        
        // Local functions

        private static VerifyCS.Test GetTestCS(string source, string corrected)
```

**Review by NewellClark — CHANGES_REQUESTED** [review]
Overall looks good.

File: src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarilyTests.cs (hunk 1 above)
Comment (wzchua) [reply src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarilyTests.cs:315]: That's not a local function

File: src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarilyTests.cs [hunk 2]
```diff
@@ -255,5 +257,82 @@ public async Task Diagnostics_VB()
...
+        {
+            var test = GetTestCS(source, corrected);
+
+            for (int i = 0; i < diagnosticResults.Length; i++)
+            {
+                var expected = diagnosticResults[i];
+                test.ExpectedDiagnostics.Add(expected);
+            }
+            await test.RunAsync();
+        }
```
Comment (Youssef1313) [inline src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarilyTests.cs:336]: Would this work?

```suggestion
        private static async Task TestCSAsync(string source, string corrected, params DiagnosticResult[] diagnosticResults)
        {
            var test = new VerifyCS.Test
            {
                TestCode = source,
                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.Preview,
                FixedCode = corrected,
            };

            test.ExpectedDiagnostics.AddRange(diagnosticResults);
            await test.RunAsync();
        }
```

File: src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarilyTests.cs [hunk 3]
```diff
@@ -1,6 +1,8 @@
 // Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
 
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeQuality.CSharp.Analyzers.QualityGuidelines;
```
Comment (buyaa-n) [inline src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarilyTests.cs:5]: NIT: I think it is redundant
```suggestion
```

**Review by buyaa-n — APPROVED** [review]
One NIT, overall LGTM, thanks!

**Pull request merged**